include/IWriteFile.h

```diff
@@ -38,7 +38,7 @@ namespace io
 
 		//! Get name of file.
 		/** \return File name as zero terminated character string. */
-		virtual const std::filesystem::path& getFileName() const = 0;
+		virtual std::string getFileName() const = 0;

```

Comment:
you do know what references are? and why we should use them?

Reply:
Yeah but std::filesystem::path::string() returns a string, not a string& so i will be returnig a ref to a temporary object

Reply:
return a `const path&`